src/cpp/rtps/reader/reader_utils.cpp

```diff
@@ -31,6 +31,12 @@ bool change_is_relevant_for_filter(
 {
     bool ret = true;
 
+    if ((change.serializedPayload.data == nullptr) &&
```

Comment:
Maybe add a comment about this precondition?

Reply:
43ae9a6